presto-hive/src/main/java/com/facebook/presto/hive/HiveLocationService.java

```diff
@@ -107,18 +110,19 @@ private boolean shouldUseTemporaryDirectory(ConnectorSession session, HdfsContex
     @Override
     public WriteInfo getQueryWriteInfo(LocationHandle locationHandle)
     {
-        return new WriteInfo(locationHandle.getTargetPath(), locationHandle.getWritePath(), locationHandle.getWriteMode());
+        return new WriteInfo(locationHandle.getTargetPath(), locationHandle.getWritePath(), locationHandle.getTmpPath(), locationHandle.getWriteMode());
     }
 
     @Override
     public WriteInfo getTableWriteInfo(LocationHandle locationHandle)
     {
-        return new WriteInfo(locationHandle.getTargetPath(), locationHandle.getWritePath(), locationHandle.getWriteMode());
+        return new WriteInfo(locationHandle.getTargetPath(), locationHandle.getWritePath(), locationHandle.getTmpPath(), locationHandle.getWriteMode());
     }
 
     @Override
     public WriteInfo getPartitionWriteInfo(LocationHandle locationHandle, Optional<Partition> partition, String partitionName)
     {
+        Path tempPath = new Path(locationHandle.getTmpPath(), randomUUID().toString().replaceAll("/", "_"));
```

Comment:
Question: why is it needed for every partition to have it's own tmp path?

Reply:
The original thought is that if we are writing to 100 different partitions, then we are effectively reducing the number of directories from 100 to 10. I'm not sure how troublesome that might be, and I'm open to remove this per-partition directory.